lib/LMS.class.php

```diff
@@ -1582,75 +1582,146 @@ public function UpdateRegisterData($name, $url, $hidden)
+		return $error->getMessage();
+	    else
+		return MSG_SENT;
+	}
+	elseif(ConfigHelper::getConfig('mail.backend') == 'phpmailer'){
+	    $this->mail_object = new PHPMailer();
+	    $this->mail_object->isSMTP();
+
+	    $this->mail_object->SMTPKeepAlive = $persist;
+
+	    $this->mail_object->Host = (!$host ? ConfigHelper::getConfig('mail.smtp_host') : $host);
+	    $this->mail_object->Port = (!$port ? ConfigHelper::getConfig('mail.smtp_port') : $port);
+	    $smtp_username = ConfigHelper::getConfig('mail.smtp_username');
+	    if (!empty($smtp_username) || $user) {
+		$this->mail_object->Username = (!$user ? $smtp_username : $user);
+		$this->mail_object->Password = (!$pass ? ConfigHelper::getConfig('mail.smtp_password') : $pass);
+		$this->mail_object->SMTPAuth  = (!$auth ? ConfigHelper::getConfig('mail.smtp_auth_type', true) : $auth);
+		$this->mail_object->SMTPSecure  = (!$auth ? ConfigHelper::getConfig('mail.smtp_secure', true) : $auth);
+	    }
+
+	    $this->mail_object->set('X-Mailer', 'LMS-' . $this->_version);
+	    if (!empty($_SERVER['REMOTE_ADDR']))
+		$this->mail_object->set('X-Remote-IP', $_SERVER['REMOTE_ADDR']);
+	    if (isset($_SERVER['HTTP_USER_AGENT']))
+		$this->mail_object->set('X-HTTP-User-Agent', $_SERVER['HTTP_USER_AGENT']);
+	    $this->mail_object->set('Mime-Version', '1.0');
+
+	    preg_match('/^(.+) <([a-z0-9_\.-]+@[\da-z\.-]+\.[a-z\.]{2,6})>$/A', $headers['From'], $from);
+	    $this->mail_object->setFrom($from[2], trim($from[1], "\""));
+	    $this->mail_object->Subject = $headers['Subject'];
+
+	    $debug_email = ConfigHelper::getConfig('mail.debug_email');
+	    if (!empty($debug_email)) {
+		$recipients = ConfigHelper::getConfig('mail.debug_email');
+	    }
+
+	    if (empty($headers['Date']))
+		$headers['Date'] = date('r');
+
+	    if ($files || $headers['X-LMS-Format'] == 'html') {
+		$boundary = '-LMS-' . str_replace(' ', '.', microtime());
+		$this->mail_object->set('Content-Type', "multipart/mixed;\n  boundary=\"" . $boundary . '"');
+
+		$this->mail_object->CharSet = 'UTF-8';
+
+		if ($files)
+		    while (list(, $chunk) = each($files))
+			$this->mail_object->AddAttachment($chunk['data'], $chunk['filename'],'base64', $chunk['content_type'], 'attachment');
+
+		if($headers['X-LMS-Format'] == 'html'){
+		    $this->mail_object->isHTML(true);
+		    $this->mail_object->AltBody = trans("To view the message, please use an HTML compatible email viewer");
+		    $this->mail_object->msgHTML($body);
+		}
+		else{
+		    $this->mail_object->isHTML(false);
+		    $this->mail_object->Body($body);
+		}
+	    }
+	    else {
+		$this->mail_object->set('Content-Type', 'text/plain; charset=UTF-8');
+		$this->mail_object->isHTML(false);
+		$this->mail_object->Body($body);
+	    }
+
+	    $this->mail_object->addAddress($recipients);
+
+	    if(!$this->mail_object->Send()) {
+		return "Mailer Error: " . $this->mail_object->ErrorInfo;
+	    } else {
+		return MSG_SENT;
+	    }
+	}
     }
```

Comment:
Ciężko uniknąć tutaj duplikacji kodu. w klasie PEAR wszystko jest ustawiane w $headers a w phpmailerze trzeba to wszystko ustawiać przez odpowiednie funkcje